File: LeetCode/java/624_Maximum_Distance_in_Arrays.java
import java.util.List;

class Solution {
    public int maxDistance(List<List<Integer>> arrays) {
        int answer = -1;
        int minNum = arrays.get(0).get(0);
        int maxNum = arrays.get(0).get(arrays.get(0).size() - 1);

        for (int i = 1; i < arrays.size(); i++) {
            int newDistance = Math.abs(minNum - arrays.get(i).get(arrays.get(i).size() - 1));

            if (answer < newDistance) {
                answer = newDistance;
            }

            newDistance = Math.abs(arrays.get(i).get(0) - maxNum);

            if (answer < newDistance) {
                answer = newDistance;
            }

            if (minNum > arrays.get(i).get(0)) {
                minNum = arrays.get(i).get(0);
            }

            if (maxNum < arrays.get(i).get(arrays.get(i).size() - 1)) {
                maxNum = arrays.get(i).get(arrays.get(i).size() - 1);
            }
        }

        return answer;
    }
}

File: LeetCode/cpp/624_Maximum_Distance_in_Arrays.cpp
class Solution {
public:
	int maxDistance(vector<vector<int>>& arrays) {
		int answer = -1;
		int minNum = arrays[0][0], maxNum = arrays[0][arrays[0].size() - 1];

		for (int i = 1; i < arrays.size(); i++) {
			int newDistance = abs(minNum - arrays[i][arrays[i].size() - 1]);

			if (answer < newDistance) {
				answer = newDistance;
			}

			newDistance = abs(arrays[i][0] - maxNum);

			if (answer < newDistance) {
				answer = newDistance;
			}

			if (minNum > arrays[i][0]) {
				minNum = arrays[i][0];
			}

			if (maxNum < arrays[i][arrays[i].size() - 1]) {
				maxNum = arrays[i][arrays[i].size() - 1];
			}
		}

		return answer;
	}
};